### PR TITLE
feed_meta Should include baseurl

### LINKED ISSUE #79: Incorrect `link[href]` value if baseurl is provided in _config.yml
Given the following `_config.yml` file content:

```yml
baseurl: /blog
feed:
  path: feed.xml
```

The liquid tag `{% feed_meta %}` renders:

```
<link type="application/atom+xml" rel="alternate" href="feed.xml" title="..." />
```

(instead of)

```
<link type="application/atom+xml" rel="alternate" href="/blog/feed.xml" title="..." />
```

Whereas the feed is properly generated at `localhost:4000/blog/feed.xml` (and not at localhost:4000/feed.xml`.

If I change the value of the `path` to `blog/feed.xml`, the `link[href]` value is correct but then the feed content is not rendered at the given URL.

Is there anything special to do? I thought we'd have followed the README correctly.

Thanks :-)

refs sudweb/blog#10

### PR DESCRIPTION
**Pull request by pathawks**
Fixes #79

### CODE REVIEW

File: lib/jekyll-feed.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def path
 
     def url
       if config["url"]
-        config["url"]
+        config["url"] + config["baseurl"]
```
Comment (benbalter) [inline lib/jekyll-feed.rb]: Would it be more robust to use URI.join here?

Comment (pathawks) [inline lib/jekyll-feed.rb]: Good call :+1:

**Comment by benbalter** [comment]
@pathawks did we lose the test when you force pushed from a375d8b to b56c3a3? Last, do we need to test with and without a base url to ensure, e.g., we don't get double trailing slashes?

**Comment by pathawks** [comment]
> @pathawks did we lose the test

Eek! Sorry! Thanks for catching that

> do we need to test with and without a base url

We already have [a test without base url](https://github.com/jekyll/jekyll-feed/blob/v0.4.0/spec/jekyll-feed_spec.rb#L187-L193). Perhaps I should add this test to the `"feed meta"` context instead of the `"with a baseurl"` context? How can we organize this to make the most sense?

**Comment by benbalter** [comment]
> We already have a test without base url. Perhaps I should add this test to the "feed meta" context instead of the "with a baseurl" context? How can we organize this to make the most sense?

Nah, just misread the diff. This is awesome. Thanks for tracking this down and putting this together. :+1:

**Comment by pathawks** [comment]
This might break some things.

Looking at [the source](https://github.com/jekyll/jekyll-feed/blob/v0.4.0/lib/jekyll-feed.rb#L21-L40), if somebody specifies a custom path for the feed, it does *not* currently respect `baseurl`. I believe it *should*, but we will have to make sure to let folks know that if the are upgrading and have a custom `feed.path` *and* `baseurl`, they will need to update the `feed.path`

**Pull request merged**